r1cs/gadgets/std/src/fields/nonnative/params.rs

```diff
@@ -1,13 +1,31 @@
+use algebra::{FpParameters, PrimeField};
 use crate::fields::nonnative::NonNativeFieldParams;
 
 /// The surfeit used for finding the optimal parameters
 pub(crate) const SURFEIT:u32 = 10;
 
 /// Obtain the parameters from a `ConstraintSystem`'s cache or generate a new one
 #[must_use]
-pub const fn get_params(target_field_prime_length: usize, base_field_prime_length: usize, pseudo_mersenne_const: Option<u64>) -> NonNativeFieldParams {
+pub fn get_params<SimulationF: PrimeField, ConstraintF: PrimeField>() -> NonNativeFieldParams {
```

Comment:
Perhaps we can make this constant too if we pass a boolean flag telling if SimulationF is a pseudo mersenne field or not (not that important, it's not much of a overhead anyway)

Reply:
It can be done but I thought we would like to remove any detail on pseudo-mersenne from publicly exposed functions